datamol/align.py

```diff
@@ -0,0 +1,252 @@
+            will emphasize the atoms that do not match, whereas AlignMol will not.
+        remove_confs: Whether to remove all conformation in the input molecule first.
+            You can set this to true when not using depiction
+        auto_select_coord_gen: Whether to automatically select the coordinate generation method.
+
+    Returns:
+        mol: aligned molecule.
+    """
+
+    if isinstance(mol, str):
+        _mol = dm.to_mol(mol)
+    elif copy:
+        _mol = dm.copy_mol(mol)
+    else:
+        _mol = mol
+
+    if _mol is None:
+        return None
+
+    if isinstance(template, str):
+        _template = dm.to_mol(template)
+    elif copy:
+        _template = dm.copy_mol(template)
+    else:
+        _template = template
+
+    if _template is None:
+        return _mol
+
+    if remove_confs:
+        _mol.RemoveAllConformers()
+
+    # EN: to make this more general and robust, we need to first check whether the template
+    # has 2D coordinates or not. If it does not, we need to compute them.
+    # This is a very rare edge case if requests are comming from the UI, but better to check than not xD
+    if _template.GetNumConformers() == 0:
+        _template = compute_2d_coords(_template)
+
+    # EN: now we can align the molecule to the template
+    # but first, we should avoid MCS as much as possible, because it's expensive
+    # so if the template is a subgraph of the molecule, no need to perform any MCS
+    # Another reason for this, is to avoid inconsistency in alignment between molecules that are
+    # supergraph of the template vs molecules that are subgraph of the template.
+    pattern = _template
+    if not _mol.HasSubstructMatch(_template):
+
+        pattern = None
+        mcs_smarts = dm.find_mcs([_mol, _template])
+
+        if mcs_smarts is not None:
+            pattern = dm.from_smarts(mcs_smarts)
+
+    if pattern is not None:
+
+        if auto_select_coord_gen:
+            rdDepictor.SetPreferCoordGen(use_depiction)
+
+        # we would need to compute 2d coordinates for the molecules if it doesn't have any
+        if _mol.GetNumConformers() == 0:
+            _mol = compute_2d_coords(_mol)
+
+        if use_depiction:
+            rdDepictor.GenerateDepictionMatching2DStructure(
+                _mol,
+                reference=_template,
+                refPatt=pattern,
+                acceptFailure=True,
+                allowRGroups=True,
+            )
+        else:
+            query_match = _mol.GetSubstructMatch(pattern)
+            template_match = _template.GetSubstructMatch(pattern)
+            rdMolAlign.AlignMol(_mol, _template, atomMap=list(zip(query_match, template_match)))
+
+    return _mol
+
+
+def auto_align_many(
+    mols: Union[Sequence[Mol], pd.Series],
+    partition_method: str = "anon-scaffold",
+    copy: bool = True,
+    cluster_cutoff: float = 0.7,
+    **kwargs,
+):
```

Comment:
@maclandrol @hadim I wonder if `should_align` shouldn't be internal to this method as an argument named `preserve_confs=True` as a default.

the internals of `should_align` is a bit an arbitrary judgement of whether or not you should align a list based on our current use case. Or renaming `should_align` to `has_confs` would make more sense, and can be used the same way. Let me know what you think.

Reply:
Good point. `should_align` is ill-defined and too much opinionated (why `any` instead of `all` for example?).

I think we should just delete it and let the user take care of this.

Do you mind not using `should_align` at all on our internal package and I will delete it later from datamol. I suggest you just use your own implementation as it was before.

Same comment for `preserve_confs`, it's ill-defined so I prefer not including it here.

Reply:
xref https://github.com/datamol-org/datamol/issues/94